lib/containers/AdvertisingParams.jsx

```diff
@@ -0,0 +1,141 @@
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* eslint react/forbid-prop-types: off */
+/* eslint react/require-default-props: off */
+
+'use strict';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import * as AdvertisingActions from '../actions/advertisingActions';
+import AdvertisingParamsControl from '../components/AdvertisingParamsControl';
+
+class AdvertisingParams extends React.PureComponent {
+    state = {
+        advParams: null,
+    };
+
+    handleAdvParamsChange = advParams => this.setState({ advParams });
+
+    handleApply() {
+        const { setAdvParams, hideAdvParamDialog } = this.props;
+        const { advParams } = this.state;
+
+        if (!advParams || !advParams.interval || !advParams.timeout) return;
+
+        setAdvParams(advParams);
+        hideAdvParamDialog();
+    }
+
+    handleCancel() {
+        const { hideAdvParamDialog } = this.props;
+        hideAdvParamDialog();
+    }
+
+    render() {
+        const { showAdvParams, advParams } = this.props;
+
+        return (
+            <Modal
+                className="adv-param-modal"
+                show={showAdvParams}
+                onHide={() => {}}
```

Comment:
Do we need to have this no-operation onHide?

Reply:
Should it perhaps call handleCancel like the Cancel button does?

Reply:
Removing it, I get the error "_this.props.onHide is not a function".
It happens when clicking outside the Modal
With handleCancel in it, it closes the Modal when clicking around it. The other modals (Advertising setup and security parameters) does not close when clicking outside, but I find it logical that it should close so it would be logical to have handleCancel there.

Reply:
I agree, I prefer being able to close the modal when clicking outside as long as there isn't any critical issue that requires action.